### PR TITLE
Documented pulp_file bindings client

### PR DESCRIPTION
**Pull request by fabricio-aguiar**
https://pulp.plan.io/issues/6518
closes #6518

### CODE REVIEW

**Comment by daviddavis** [comment]
Thanks for the PR. How will these docs get updated whenever there is a change to the API? Will it be up to developers to remember to regenerate these docs when they make a change?

File: docs/pulp_file-client.rst [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+   * - *ContentFilesApi*
+     - `\ **list** <pulp_file-client/ContentFilesApi.md#list>`_
+     - **GET** /pulp/api/v3/content/file/files/
+     - List file contents
+   * - *ContentFilesApi*
+     - `\ **read** <pulp_file-client/ContentFilesApi.md#read>`_
+     - **GET** {file_content_href}
+     - Inspect a file content
+   * - *DistributionsFileApi*
+     - `\ **create** <pulp_file-client/DistributionsFileApi.md#create>`_
```
Comment (fabricio-aguiar) [inline docs/pulp_file-client.rst:117]: Initially, it breaks here due to 
```
Warning, treated as error:
/home/vagrant/devel/pulp_file/docs/pulp_file-client.rst:3:Duplicate explicit target name: "**create**".

```

File: docs/pulp_file-client/ContentFilesApi.rst [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+
+Create a file content
+
+Trigger an asynchronous task to create content,optionally create new repository version.
+
+Example
+^^^^^^^
+
+
+* Basic Authentication (Basic):
```
Comment (fabricio-aguiar) [inline docs/pulp_file-client/ContentFilesApi.rst:40]: then it breaks here with
```
Warning, treated as error:
/home/vagrant/devel/pulp_file/docs/pulp_file-client/ContentFilesApi.rst:40:Inline literal start-string without end-string.
```

File: docs/pulp_file-client/ContentFilesApi.rst [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+  ```python
```
Comment (fabricio-aguiar) [inline docs/pulp_file-client/ContentFilesApi.rst:41]: If I remove the * from the previous line it breaks here with:
```
Warning, treated as error:
/home/vagrant/devel/pulp_file/docs/pulp_file-client/ContentFilesApi.rst:41:Inline literal start-string without end-string.
```

**Comment by fabricio-aguiar** [comment]
> Thanks for the PR. How will these docs get updated whenever there is a change to the API? Will it be up to developers to remember to regenerate these docs when they make a change?

@dkliban said it would be manual due to readthedocs, but after trying to build the docs, I suspect `m2r` has some limitations and making these docs compatible with sphinx will need too many hacks

**Pull request closed without merging**